### PR TITLE
Added redux  to react: store, actions, reduser, ProtectedRoute

### PR DESCRIPTION
**Pull request by YahorKazlou**

### CODE REVIEW

File: src/components/common/Input.jsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { useState } from 'react';
+import styles from './Input.module.css';
+
+const Input = ({ onChange, ...otherProps }) => {
+    const [value, setValue] = useState('');
+
+    const onChangeValue = (e) => {
+        const value = e.target.value;
+        setValue(value);
+        onChange?.(value);
```
Comment (Den4ik5) [inline src/components/common/Input.jsx]: what is it?

Comment (Den4ik5) [reply src/components/common/Input.jsx]: what have are you passing to the Input?

Comment (Den4ik5) [reply src/components/common/Input.jsx]: please use propTypes here in the input

File: src/components/common/Input.jsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import { useState } from 'react';
+import styles from './Input.module.css';
+
+const Input = ({ onChange, ...otherProps }) => {
+    const [value, setValue] = useState('');
+
+    const onChangeValue = (e) => {
```
Comment (Den4ik5) [inline src/components/common/Input.jsx]: bad method naming

File: src/pages/LoginPage.jsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
+import { useState } from 'react';
+import styles from '../App.module.css';
+import Input from '../components/common/Input';
+import { useDispatch } from 'react-redux';
+import { login as loginAction } from '../store/user/actions';
+import Logo from '../components/Navigation/Logo';
+
+const USER_CREDENTIALS = {
```
Comment (Den4ik5) [inline src/pages/LoginPage.jsx]: please consider storing constants in a separate file

File: src/store/store.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import { createStore, combineReducers } from 'redux';
+
+import { composeWithDevTools } from '@redux-devtools/extension';
+import userReducer from './user/reducer';
+
+const store = createStore(
+    combineReducers({
```
Comment (Den4ik5) [inline src/store/store.js]: why do we need combineReducers here? you are using one at the moment

File: src/store/user/actions.js [hunk 5]
```diff
@@ -0,0 +1,5 @@
+export const USER_LOGIN = 'USER/LOGIN';
+export const login = (userData) => ({
+    type: USER_LOGIN,
```
Comment (Den4ik5) [inline src/store/user/actions.js]: leave just a LOGIN

File: src/components/common/Input.jsx [hunk 6]
```diff
@@ -0,0 +1,25 @@
+import { useState } from 'react';
+import styles from './Input.module.css';
+
+const Input = ({ onChange, ...otherProps }) => {
```
Comment (Den4ik5) [inline src/components/common/Input.jsx]: why do we need this otherProps, what for?

File: src/components/common/Input.jsx (hunk 2 above)
Comment (YahorKazlou) [reply src/components/common/Input.jsx:7]: Done

File: src/store/store.js (hunk 4 above)
Comment (YahorKazlou) [reply src/store/store.js]: I made it just in case for the future, but right now it's not needed. Done

File: src/store/user/actions.js (hunk 5 above)
Comment (YahorKazlou) [reply src/store/user/actions.js]: Done

File: src/components/common/Input.jsx (hunk 6 above)
Comment (YahorKazlou) [reply src/components/common/Input.jsx]: We pass props like placeholder and type into the <input /> without changes.

File: src/components/common/Input.jsx (hunk 1 above)
Comment (YahorKazlou) [reply src/components/common/Input.jsx]: Done. 
I added propTypes to make it clear what I'm passing to the Input

File: src/pages/LoginPage.jsx (hunk 3 above)
Comment (YahorKazlou) [reply src/pages/LoginPage.jsx]: Done

File: src/components/common/Input.jsx (hunk 2 above)
Comment (YahorKazlou) [reply src/components/common/Input.jsx]: What would you suggest?

File: src/pages/LoginPage.jsx (hunk 3 above)
Comment (Den4ik5) [reply src/pages/LoginPage.jsx]: please create index.js in constants folder and export everything from there

File: src/store/store.js (hunk 4 above)
Comment (Den4ik5) [reply src/store/store.js]: No need as long as it's not used

File: src/pages/LoginPage.jsx (hunk 3 above)
Comment (YahorKazlou) [reply src/pages/LoginPage.jsx]: Done

**Pull request merged**